File: app/src/main/java/com/example/anton/sprintcalendar/TeamMember.java
package com.example.anton.sprintcalendar;

import com.google.common.base.MoreObjects;

import org.joda.time.LocalDate;

import java.util.HashMap;
import java.util.Map;

public class TeamMember {
    private String name;
    private Map<LocalDate, PresenceType> absenceMap = new HashMap<>();

    public TeamMember(String name) {
        this.name = name;
    }


    public String getName() {
        return name;
    }

    public void setPresence(LocalDate localDate, PresenceType presenceType) {
        absenceMap.put(localDate, presenceType);
    }

    public PresenceType presence(LocalDate date) {
        PresenceType presenceType = absenceMap.get(date);
        return PresenceType.safePresence(presenceType);
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("name", name)
                .toString();
    }
}

File: app/src/main/kotlin/net/sourcefusion/agiletools/sprintcalendar/TeamMember.kt
package net.sourcefusion.agiletools.sprintcalendar

import org.joda.time.LocalDate

data class TeamMember(val name: String, var id:Long? = null) {

    private val absenceMap = hashMapOf<LocalDate, PresenceType>()

    fun setPresence(localDate: LocalDate, presenceType: PresenceType) {
        absenceMap[localDate] = presenceType
    }

    fun presence(date: LocalDate): PresenceType {
        val presenceType = absenceMap[date]
        return PresenceType.safePresence(presenceType)
    }

    fun presenceIterator() = absenceMap.iterator()
}
